onnx/defs/generator/defs.cc

```diff
@@ -523,4 +527,187 @@ ONNX_OPERATOR_SET_SCHEMA(
           updateOutputShape(ctx, 0, {batch_size, sample_size});
         }));
 
+static const char* Range_ver11_doc = R"DOC(
+Generate a tensor containing a sequence of numbers that begin at `start` and extends by increments of `delta` 
+up to `limit` (exclusive).)DOC";
+
+template <typename T>
+inline int compute_output_dim_for_range(
+    const TensorProto* start,
+    const TensorProto* limit,
+    const TensorProto* delta) {
+  const auto& start_data = ParseData<T>(start);
+  const auto& limit_data = ParseData<T>(limit);
+  const auto& delta_data =
+      delta != nullptr ? ParseData<T>(delta) : std::vector<T>(1, 1);
+
+  if (start_data.size() != 1 || limit_data.size() != 1 ||
+      delta_data.size() != 1) {
+    fail_shape_inference(
+        "Input to 'Range' op should be scalars (Tensor with only one element)");
+  }
+
+  int n = static_cast<int>(
+      ceil((1.0 * (limit_data[0] - start_data[0])) / delta_data[0]));
+  if (n < 0)
+    n = 0;
+
+  return n;
+}
+
+const std::vector<NodeProto> build_nodes_range_op() {
+    // body for 'Loop node'
+    GraphProto loop_sub_graph;
+    loop_sub_graph.set_name("loop body attribute");
```

Comment:
Please use a name like loop_body_attribute (a valid C identifier). See: https://github.com/onnx/onnx/blob/master/docs/IR.md#names-within-a-graph … while this check is probably not done (because of some models that violate this), it will help us in the future when we start enforcing these requirements.

Reply:
Done